### PR TITLE
add AWS SAM support

### PR DESCRIPTION
**Pull request by gruebel**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

- adjusted Lambda checks to support Serverless resource types
- adjusted API Gateway checks to support Serverless resource types
- SAM `Globals` have specific rules on property overrides https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification-template-anatomy-globals.html#sam-specification-template-anatomy-globals-overrideable
- actually not all properties can be overridden with `Globals`, but I didn't want to maintain the list, which is growing with new features added to SAM

### CODE REVIEW

File: checkov/cloudformation/checks/resource/aws/LambdaDLQConfigured.py [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
+        properties = conf.get("Properties")
+        if properties:
+            self.evaluated_keys = ["Properties/DeadLetterConfig/TargetArn"]
+            dlc = properties.get("DeadLetterConfig")
+            if dlc and dlc.get("TargetArn"):
+                return CheckResult.PASSED
 
-    def get_expected_value(self) -> str:
-        return ANY_VALUE
+            self.evaluated_keys.append("Properties/DeadLetterQueue/TargetArn")
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/LambdaDLQConfigured.py:22]: @gruebel Why append and not just set the evaluated keys to this?

File: checkov/common/graph/graph_builder/graph_components/blocks.py [hunk 2]
```diff
@@ -44,7 +44,10 @@ def _extract_inner_attributes(self) -> Dict[str, Any]:
             if isinstance(attribute_value, dict) or (
                     isinstance(attribute_value, list) and len(attribute_value) > 0 and isinstance(attribute_value[0],
                                                                                                   dict)):
-                inner_attributes = get_inner_attributes(attribute_key, attribute_value)
+                inner_attributes = self.get_inner_attributes(
+                    attribute_key=attribute_key,
+                    attribute_value=attribute_value,
+                )
```
Comment (nimrodkor) [inline checkov/common/graph/graph_builder/graph_components/blocks.py:50]: @gruebel Unnecessary change?

**Review by nimrodkor — APPROVED** [review]
Very, very nice. 👏  @gruebel

File: checkov/common/graph/graph_builder/graph_components/blocks.py (hunk 2 above)
Comment (gruebel) [reply checkov/common/graph/graph_builder/graph_components/blocks.py:50]: before `get_inner_attributes` was a function inside the file, now it is a `@classmethod` that's why there is now a `self.` at the beginning and it is reformated, because I implemented it differently before and them came up with a better approach 😄

File: checkov/cloudformation/checks/resource/aws/LambdaDLQConfigured.py (hunk 1 above)
Comment (gruebel) [reply checkov/cloudformation/checks/resource/aws/LambdaDLQConfigured.py:22]: in general I was not sure, how to best handle `evaluated_keys` here, because it actually depends on the resource type. Do we want to add it as an attribute when the base class is invoked and passed the resource type?

**Review by nimrodkor — APPROVED** [review]

**Comment by schosterbarak** [comment]
very nice. please update README.md too :)
